### PR TITLE
Replace whacko with JSDOM, remove $

### LINKED ISSUE #392: whacko is no longer maintained.

### PR DESCRIPTION
**Pull request by jennyliang220**
Fixes #392.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,8 @@
     "express": "4.17.1",
     "express-handlebars": "4.0.4",
     "insafe": "0.5.0",
+    "jquery": "3.5.1",
```
Comment (deniak) [inline package.json:16]: remove jquery and whacko dependency

File: lib/rules/sotd/process-document.js [hunk 2]
```diff
@@ -30,29 +30,28 @@ exports.check = function (sr, done) {
...
-            if (sr.norm($p.text()) === boilerplate && $p.find("a").attr("href") === procUri) {
+        sotd.querySelectorAll("p").forEach(function (p) {
+            if (sr.norm(p.textContent) === boilerplate && p.querySelector("a").getAttribute("href") === procUri) {
                 if (found) sr.error(self, "multiple-times", { process: proc });
                 else {
                     found = true;
                 }
             }
-            else if (deprecatedBoilerplate === sr.norm($p.text()) || deprecatedUri === $p.find('a').attr('href')) {
+            else if (deprecatedBoilerplate === sr.norm(p.textContent) || deprecatedUri === p.querySelector('a').getAttribute('href')) {
```
Comment (deniak) [inline lib/rules/sotd/process-document.js:44]: `find()` returns a new Cheerio object whether the selector matches some elements or not. However, `querySelector` will return `null` if no element can be found so you need to check if it returns `null` (everywhere where we used `find`) before getting the attribute.
At the moment, an exception is thrown if you try to check a document.

File: lib/validator.js [hunk 3]
```diff
@@ -232,83 +234,86 @@ Specberus.prototype.getDocumentDateElement = function () {
...
+        return true;
     });
-    if (!$startH2)
+    if (!startH2)
         this.sotdSection = null;
     else {
         var started = false;
-        $startH2.parent().children().each(function () {
-            if ($startH2[0] === this) {
+        Array.prototype.every.call(startH2.parentElement.children, function (element) {
```
Comment (deniak) [inline lib/validator.js:259]: `s/every/forEach`

File: lib/validator.js [hunk 4]
```diff
@@ -353,17 +358,14 @@ Specberus.prototype.getDelivererIDs = function () {
...
-
-            var item = $(this)
-            ,   href = item.attr('href')
-            ,   text = self.norm(item.text())
+        , self = this
+        , sotd = self.getSotDSection()
+        , sotdLinks = sotd && sotd.querySelectorAll('a[href]');
+
+    if (ids.length === 0 && sotdLinks.length > 0) {
+        Array.prototype.every.call(sotdLinks, function (item) {
```
Comment (deniak) [inline lib/validator.js:366]: `s/every/forEach`

File: lib/rules/sotd/stability.js [hunk 5]
```diff
@@ -3,11 +3,11 @@
...
 // inappropriate to cite this document as other than work in progress.</p>
-function findSW ($candidates, sr) {
-    var $sw = null;
-    $candidates.each(function () {
-        var $p = sr.$(this)
-        , text = sr.norm($p.text())
+const util = require('../../util');
+function findSW (candidates, sr) {
+    var sw = null;
+    Array.prototype.every.call(candidates, function (p) {
```
Comment (deniak) [inline lib/rules/sotd/stability.js:9]: `s/every/forEach`

File: lib/rules/sotd/supersedable.js [hunk 6]
```diff
@@ -14,20 +15,28 @@ const self = {
...
 exports.check = function (sr, done) {
-    var $sotd = sr.getSotDSection();
-    if ($sotd && $sotd.length) {
-        var $em = $sotd.filter("p").children("em").first();
-        if (!$em.length) $em = $sotd.find("p").children("em").first();
-        var txt = sr.norm($em.text())
+    var sotd = sr.getSotDSection();
+    if (sotd) {
+        var em;
+        util.filter(sotd, "p").every(function (paragraph) {
```
Comment (deniak) [inline lib/rules/sotd/supersedable.js:21]: `s/every/forEach`

File: lib/rules/sotd/submission.js [hunk 7]
```diff
@@ -6,11 +7,10 @@ const self = {
...
 exports.name = self.name;
 
-function findSubmText ($candidates, sr) {
-    var $st = null;
-    $candidates.each(function () {
-        var $p = sr.$(this)
-        ,   text = sr.norm($p.text())
+function findSubmText (candidates, sr) {
+    var st = null;
+    Array.prototype.every.call(candidates, function (p) {
```
Comment (deniak) [inline lib/rules/sotd/submission.js:12]: `s/every/forEach`

**Comment by tripu** [comment]
:clap: :tada:  
:pray:

**Pull request merged**